lib/Sema/ConstraintSystem.cpp

```diff
@@ -2765,6 +2765,9 @@ bool ConstraintSystem::diagnoseAmbiguityWithFixes(
     ParameterList,
     /// General ambiguity failure.
     General,
+    /// Argument mismatch ambiguity where each solution has the same
+    /// argument mismatch fixes for the same call.
+    ArgumentMismatch
```

Comment:
Do you mean the same callee location? because in those cases all the fixes are the same, but each one for an argument. e.g. In the case of 
```swift
func f<T>(_: T, _: T) {}
f(Int(1), Float(1))
```
There is a fix located at arg #1 in the first solution which was attempted `(Int, Int)->Void` and another at #0

Reply:
Yeah, the two fixes here have different locators so it's not quite the same case as N common fixes across N solutions

Reply:
Actually, maybe the locator doesn't matter. With my changes [here](https://github.com/apple/swift/compare/master...hborla:ambiguity-with-fixes?expand=1), I've implemented a special method on `CSFix` for diagnosing for ambiguity when the fix is "common" across all solutions, but I consider a fix "common" only when the locator is the same. Instead of assuming that the locator must be the same, we can let the particular fix kind decide what it means to be "common"

Reply:
Humm ... so in for these case for `AllowTupleTypeMismatch` and `AllowArgumentMismatch` fixes will be special cases where we could ignore locator for common to be able to disambiguate.